### DISCUSSION TITLE
setup DataDefinitions with pimcore 10.4

### DISCUSSION THREAD

**Discussion by sandyHM** [pimcore/pimcore#15268] [Q&A / Support]
how to setup DataDefinitions with pimcore 10.4

**Comment by dpfaffenbauer** [comment]
`composer require w-vision/data-definitions`

**Reply by sandyHM** [reply]
thank you @dpfaffenbauer ,
it install data-definitions v1.* for pimcore 10.4 and it does not work with pimcore 10.4  . i need to install data-definitions for pimcore v10.4

**Reply by dpfaffenbauer** [reply]
try it with `composer require w-vision/data-definitions:dev-master`

**Reply by sandyHM** [reply]
Problem 1
    - Root composer.json requires w-vision/data-definitions dev-master -> satisfiable by w-vision/data-definitions[dev-master].
    - w-vision/data-definitions dev-master requires pimcore/pimcore ^10.5 -> found pimcore/pimcore[v10.5.0, ..., v10.6.1] but it conflicts with your root composer.json require (v10.4.3).

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions

**Reply by dpfaffenbauer** [reply]
oh yeah, right, most of the plugins only work with Pimcore 10.5 and upwards. I recommend you to update to 10.5

**Reply by sandyHM** [reply]
thank you  vary much @dpfaffenbauer 
but i do not want upgrade pimcore 10.4 to 10.5
